### ISSUE TITLE
Xorg module libvnc.so needs documentation

### ISSUE THREAD

**Issue by pprindeville** [TigerVNC/tigervnc#394]
There's documentation for vncviewer, vncserver, x0vncserver, vncconfig... but nothing describing the "vnc" module.

**Comment by CendioOssman** [comment]
Indeed. Much of the documentation for Xvnc is valid for libvnc.so as well though.

**Comment by Pascal666** [comment]
Anyone figure out what libvnc.so is yet?

**Comment by toff** [comment]
This works for me on an Ubuntu based system (18.04).
To install the package and create a password file for root in `/root/.vnc/passwd` :
```
sudo -i
apt install tigervnc-xorg-extension
vncpasswd
```
Then from the [Arch wiki](https://wiki.archlinux.org/index.php/TigerVNC#Expose_the_local_display_directly)

Create the file : `/etc/X11/xorg.conf.d/10-vnc.conf`
```
Section "Module"
  Load "vnc"
EndSection

Section "Screen"
  Identifier "Screen0"
  Option "UserPasswdVerifier" "VncAuth"
  Option "PasswordFile" "/root/.vnc/passwd"
EndSection
```
And restart your Xorg session.

You Xorg server should now be listening on port 5900.
You can check this with :
`sudo ss -nlpt | grep 5900`
`LISTEN    0         5                  0.0.0.0:5900             0.0.0.0:*        users:(("Xorg",pid=25921,fd=22))                                               `

And you can connect to it with your vnc client.

**Comment by herreradimas** [comment]
I tried run module VNC but this did work, here specs:

OS:
Linux debian 4.19.0-13-amd64 #1 SMP Debian 4.19.160-2 (2020-11-28) x86_64 GNU/Linux

GRAPHICS CARD:
01:00.0 VGA compatible controller: NVIDIA Corporation GT218M [NVS 3100M] (rev a2) (prog-if 00 [VGA controller])
	Subsystem: Hewlett-Packard Company GT218M [NVS 3100M]
	Flags: bus master, fast devsel, latency 0, IRQ 38
	Memory at d2000000 (32-bit, non-prefetchable) [size=16M]
	Memory at c0000000 (64-bit, prefetchable) [size=256M]
	Memory at d0000000 (64-bit, prefetchable) [size=32M]
	I/O ports at 5000 [size=128]
	[virtual] Expansion ROM at 000c0000 [disabled] [size=128K]
	Capabilities: [60] Power Management version 3
	Capabilities: [68] MSI: Enable+ Count=1/1 Maskable- 64bit+
	Capabilities: [78] Express Endpoint, MSI 00
	Capabilities: [b4] Vendor Specific Information: Len=14 <?>
	Capabilities: [100] Virtual Channel
	Capabilities: [128] Power Budgeting <?>
	Capabilities: [600] Vendor Specific Information: ID=0001 Rev=1 Len=024 <?>
	Kernel driver in use: nvidia
	Kernel modules: nvidia

**Comment by mghadam** [comment]
The [tigervnc-xorg-extension](https://packages.debian.org/bookworm/tigervnc-xorg-extension) package does not work on Debian 12 for some reason.

There is a sample config file `/usr/share/doc/tigervnc-xorg-extension/99-tigervnc.conf` with unix pam authentication support however I have not been able to utilize it so far even with a simple configuration like  the following  with a valid `/etc/vnc/passwd` file and readable 644 permissions.



```
Section "Module"
  Load "vnc"
EndSection

Section "Screen"
  Identifier "Screen0"
  Option "UserPasswdVerifier" "VncAuth"
  Option "PasswordFile" "/etc/vnc/passwd"
EndSection
```


nothing listens on port 5900 after restarting the session
